src/pm/hydra/pm/pmiserv/common.c

```diff
@@ -122,11 +122,30 @@ char *HYD_pmcd_pmi_find_token_keyval(struct HYD_pmcd_token *tokens, int count, c
 HYD_status HYD_pmcd_pmi_allocate_kvs(struct HYD_pmcd_pmi_kvs ** kvs, int pgid)
 {
     HYD_status status = HYD_SUCCESS;
+    char hostname[MAX_HOSTNAME_LEN];
+#ifdef MPL_HAVE_CLOCK_GETTIME
+    struct timespec ts;
+#else
+    struct timeval tv;
+#endif
+    int rnd;
+    unsigned int seed;
 
     HYDU_FUNC_ENTER();
 
+    gethostname(hostname, MAX_HOSTNAME_LEN);
+#ifdef MPL_HAVE_CLOCK_GETTIME
+    MPL_wtime(&ts);
+    seed = (unsigned int)ts.tv_nsec;
+#else
+    gettimeofday(&tv, NULL);
+    seed = (unsigned int)tv.tv_usec;
+#endif
```

Comment:
I think it would be cleaner to only use MPL, and convert it to double-precision seconds.  Then you can convert it to microseconds and cast to `unsigned int`.

Reply:
I have modified this to use MPL_time_t as you have suggested.